tests/functional/conftest.py

```diff
@@ -44,13 +57,27 @@ def wait_for_maximum_list_length(
     ) -> None:
         """Wait for the list() length to be no greater than expected maximum or fail
         test if timeout is exceeded"""
-        for _ in range(max_iterations):
-            if len(rest_manager.list()) <= max_length:
+        logging.info(f"Checking {description!r} has no more than {max_length} items")
+        for count in range(max_iterations):
+            items = rest_manager.list()
+            if len(items) <= max_length:
                 break
+            logging.info(
+                f"Iteration: {count} Waiting for {description!r} items to be deleted: "
+                f"{[x.name for x in items]}"
+            )
             time.sleep(SLEEP_INTERVAL)
-        assert len(rest_manager.list()) <= max_length, (
-            f"Did not delete required items for {description}. "
-            f"Elapsed_time: {time.perf_counter() - start_time}"
+
+        items = rest_manager.list()
```

Comment:
Don't we already have it from inside the loop before the break?

Reply:
Good catch!